src/SimpleSimulator.cpp

```diff
@@ -425,15 +454,52 @@ int main(int argc, char **argv) {
     /* Create a workload */
     std::cerr << "Constructing workload specification..." << std::endl;
 
-    auto workload_spec = fill_streaming_workflow(
-        num_jobs, infiles_per_job,
-        average_flops, sigma_flops,
-        average_memory,sigma_memory,
-        average_infile_size, sigma_infile_size,
-        average_outfile_size, sigma_outfile_size
-    );
+    std::map<std::string, JobSpecification> workload_spec = {};
 
-    std::cerr << "The workflow has " << std::to_string(num_jobs) << " unique jobs" << std::endl;
+
+    if(workflow_configurations.size() == 0){
+        std::string workflow_type_lower = boost::to_lower_copy(vm["workflow-type"].as<std::string>());
+        workload_spec = fill_workflow(
+            num_jobs, infiles_per_job,
+            average_flops, sigma_flops,
+            average_memory,sigma_memory,
+            average_infile_size, sigma_infile_size,
+            average_outfile_size, sigma_outfile_size,
+            get_workflow_type(workflow_type_lower), ""
+        );
+
+        std::cerr << "The workflow has " << std::to_string(num_jobs) << " unique jobs" << std::endl;
+    }
+    else {
+        for(auto &wf_confpath : workflow_configurations){
+            std::ifstream wf_conf(wf_confpath);
+            nlohmann::json wf_json = nlohmann::json::parse(wf_conf);
+
+            // Checking json syntax to match workflow spec
+            for (auto &wf_key : workflow_keys){
+                try {
+                    if(!wf_json.contains(wf_key)){
+                        throw std::invalid_argument("ERROR: the workflow configuration " + wf_confpath + " must contain " + wf_key + " as information.");
+                    }
+                }
+                catch(std::invalid_argument& e){
+                    std::cerr << e.what() << std::endl;
+                    return -1;
```

Comment:
Typedef `EXIT_FAILURE` might be easier here for the someone reading the code.

Reply:
Do you have an example, how to do what? I think I don't understand exactly what you mean.

Reply:
`EXIT_FAILURE` is just a C++ builtin: 
```C++
#define EXIT_FAILURE 1
```
Just return this, instead of `-1`.

Reply:
Done: https://github.com/HEPCompSim/DCSim/pull/41/commits/2af52b3808e9a4e155865602df1d76cb7945d897